### PR TITLE
AB#83312 - Improve the width of history view

### PR DESCRIPTION
**Pull request by guisoares1**
# Description

Implemented the option to dynamically resize the history screen.

## Useful links

- [Link to ticket](https://dev.azure.com/whohq/emssafe/_workitems/edit/83312)

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)
# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] Changed the size of the history screen and observed behavior.

##Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/55603259/a995d327-90da-4d74-a935-846dc15e45e2

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: libs/shared/src/lib/components/record-history/record-history.component.html [hunk 1]
```diff
@@ -1,150 +1,153 @@
...
-      <div class="flex items-center justify-between">
-        <label>
-          {{ 'common.input.dateRange' | translate }}
-        </label>
-        <ui-button
-          class="max-sm:!w-auto mb-3"
-          variant="danger"
-          (click)="clearDateFilters()"
-          *ngIf="filters.get('startDate')?.value || filters.get('endDate')?.value"
+<div appResizable>
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/record-history/record-history.component.html]: @guisoares1 
is it linked to any directive or is it useless?

I couldn't find any reference to that in the code

**Comment by AntoineRelief** [comment]
@guisoares1 
I'm putting this one back as draft
it works fine when the history is in the sidenav, it does not when the history is in a popup

please check about how history works in that case

I'd have preferred a way to enable resizing on elements like sidenav / dialog, that would could pass to some of them, so it offers more flexibility

like what you did, it could work, but it's too deeply linked to the history, I don't like this solution

File: libs/shared/src/lib/components/record-history/record-history.component.html (hunk 1 above)
Comment (guisoares1) [reply libs/shared/src/lib/components/record-history/record-history.component.html]: it useless, i forgot to take this off

**Comment by guisoares1** [comment]
@AntoineRelief  I'll explain what I'm trying to do (it's not finished yet.), if you don't agree with this path, let's go back to the analysis part.

- Creation of a component that will enable screen resizing. I will implement it in the history dialog and sidenav.
- This should dynamically resize the screen, with a limit on how far the user can go.

I tried a simpler way, changing the css, html and js but I didn't get a satisfactory result.

**Comment by AntoineRelief** [comment]
@guisoares1 
okay for that
instead of using a component, I think you could use a directive
but if you manage to do it with a component, fine for me

**Comment by guisoares1** [comment]
**Summary of what was done:**
- Created logic for resizing to work on dialog screens and added resizing option to element edges.
- Modified the location of the resize component, in order to correct the cyclic dependency error.

**Result**

https://github.com/ReliefApplications/ems-frontend/assets/55603259/b22f7cf6-3f3c-48e1-b727-051bee0c16d8

File: libs/ui/src/lib/app-resizable-box/app-resizable-box.component.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  /** last value of y*/
+  private lastY = 0;
+  /** width of the element*/
+  private width = 0;
+  /** height of the element */
+  private height = 0;
+  /** direction of the resize */
+  private direction = '';
+  /**  Mouse move subscription*/
+  private mouseMoveSubscription: Subscription | undefined;
```
Comment (AntoineRelief) [inline libs/ui/src/lib/app-resizable-box/app-resizable-box.component.ts]: @guisoares1 for future PRs, just use the mouseMoveSubscription?: Subscription; notation

File: libs/ui/src/lib/app-resizable-box/app-resizable-box.component.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  /** Mouse up subscription */
+  private mouseUpSubscription: Subscription | undefined;
+
+  /**
+   * Constructor
```
Comment (AntoineRelief) [inline libs/ui/src/lib/app-resizable-box/app-resizable-box.component.ts]: @guisoares1 
"constructor" is not a valid jsdoc

File: libs/ui/src/lib/app-resizable-box/app-resizable-box.component.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
+import { CommonModule } from '@angular/common';
+import { Component, ElementRef, Input, OnDestroy, OnInit } from '@angular/core';
+import { Subscription, fromEvent } from 'rxjs';
+
+/**
+ *
```
Comment (AntoineRelief) [inline libs/ui/src/lib/app-resizable-box/app-resizable-box.component.ts]: @guisoares1 
not valid jsdoc, please at least put a description of what it does

**Comment by guisoares1** [comment]
@AntoineRelief Made all requested changes.

**Comment by RenzoPrats** [comment]
@AntoineRelief @guisoares1 It's working fine, but should we define how much the user can expand and how much the user can collapse? because we current could have something like this 
Captura de tela de 2024-01-17 11-27-21
Captura de tela de 2024-01-17 11-24-29
Captura de tela de 2024-01-17 11-23-49

**Comment by AntoineRelief** [comment]
@RenzoPrats 
agree

**Comment by RenzoPrats** [comment]
@AntoineRelief I did some logic to set max, min width and height, it seems to work fine. The last remaining thing is about page size change, current the calculation of width and height is 80% of page size, but if we expand max as possible and then reduce the size of the page it will not adjust automatically and can break the logic. Should we do something about that?

**Comment by AntoineRelief** [comment]
Converting to draft, it's breaking stuff, see screenshot:

Capture d’écran 2024-01-18 à 13 37 41

I'm pretty sure directives or custom directive can be made for that

see this package for example:
https://github.com/mattlewis92/angular-resizable-element

I'm surprised stuff like that can't be used

**Pull request closed without merging**